Cleaner Creature follow logic

danilopucci opened the pull request:
<!-- Note: Lines with this <!-- syntax are comments and will not be visible in
     your pull request. You can safely ignore or remove them. -->

### Pull Request Prelude
This PR adds small changes on Creature Follow logic by renaming and moving some code. The changes here does not affect any logic to avoid introducing any kind of bug.

<!-- Thank you for working on improving The Forgotten Server! -->
<!-- Please complete these steps and check the following boxes by putting an `x`
     inside the [brackets] before filing your Pull Request. -->

- [X] I have followed [proper The Forgotten Server code styling][code].
- [X] I have read and understood the [contribution guidelines][cont] before making this PR.
- [X] I am aware that this PR may be closed if the above-mentioned criteria are not fulfilled.

### Changes Proposed
Small improvement on Creature Follow logic by:

- changes on ```Creature::getPathSearchParams```
  - rename it to ```Creature::buildFindPathParams``` as the object name that it is building is ```FindPathParams```
  - remove abbreviation on its parameter (and also from the callers)
  - add a new parameter which sets ```fullSearchPath```
  
- changes on ```Creature::goToFollowCreature```:
  - its callback was renamed from ```onFollowCreatureComplete``` to ```onGoToFollowCreatureComplete```
  - Monster logic was moved to proper ```Monster::goToFollowCreature``` override
  - repeated code was moved to a new method ```Creature::updateFollowPath```

I splitted the mentioned changes on different commits, so if the reviewers wants to look into small pieces of changes, it is easier to look on each commit.

It is worth to say that proabably there is more logic that can be cleaned and simplified, but I avoid to do that now just to be a safer change.  

**Issues addressed:** <!-- Write here the issue number, if any. -->


<!-- You can safely ignore the links below:  -->

[cont]: https://github.com/otland/forgottenserver/wiki/Contributing
[code]: https://github.com/otland/forgottenserver/wiki/TFS-Coding-Style-Guide

4mrcn4 commented:
> I didn't understand why you changed the short name of the fpp variable to findPathParams XD but ok..

In time you will learn that letternamed variables does not save time, they waste it.

MillhioreBT commented:
> > I didn't understand why you changed the short name of the fpp variable to findPathParams XD but ok..
> 
> In time you will learn that letternamed variables does not save time, they waste it.

No one has said anything about that... wtf

I'm just saying that sometimes people love to modify unnecessary things, how ugly the whole name repeated everywhere looks, as if that change were necessary, it's silly to change the name of the variable, it will only add noise to the PR

danilopucci commented:
Yeah, as @4mrcn4 pointed out, I renamed it to be more readable and meaningfull to what the variable represents. When I was first reading it, it took me some time to remember what fpp was :p 

I 100% agree on @MillhioreBT thoughs on avoid change unecessary things and bring noise to the PR, but I honestly do not think that is the case, as the variable name changes are related to the method that I changed and on the diff is pretty clear that it is a single renaming.

MillhioreBT commented:
> Yeah, as @4mrcn4 pointed out, I renamed it to be more readable and meaningfull to what the variable represents. When I was first reading it, it took me some time to remember what fpp was :p
> 
> I 100% agree on @MillhioreBT thoughs on avoid change unecessary things and bring noise to the PR, but I honestly do not think that is the case, as the variable name changes are related to the method that I changed and on the diff is pretty clear that it is a single renaming.

How many hours did it take you to identify what fpp that means?
![image](https://github.com/otland/forgottenserver/assets/28090948/d0f0bfd1-835d-4afd-aa4d-f8b8216d09d8)
XD

I love that everyone supports the project and adds things, so it was just a healthy comment, thanks for the changes <3

danilopucci commented:
@MillhioreBT Thank you so much

I have just fixed the code style

danilopucci commented:
@MillhioreBT did you had any time to test it?

marmichalski commented on src/creature.cpp:
```diff
@@ -939,60 +939,39 @@ bool Creature::setAttackedCreature(Creature* creature)
+bool Creature::updateFollowPath(FindPathParams& findPathParams)
+{
+	listWalkDir.clear();
+	if (getPathTo(followCreature->getPosition(), listWalkDir, findPathParams)) {
+		hasFollowPath = true;
+	} else {
+		hasFollowPath = false;
+	}
+
+	return hasFollowPath;
```
```suggestion

	return hasFollowPath = getPathTo(followCreature->getPosition(), listWalkDir, findPathParams);
```

💩

MillhioreBT commented:
> @MillhioreBT did you had any time to test it?

Please leave the variable name as it was originally, just make the necessary logic changes to clean up the logic.

danilopucci commented:
> Please leave the variable name as it was originally, just make the necessary logic changes to clean up the logic.
I reverted the findPathParams to fpp, can you please take a look again? @MillhioreBT

danilopucci commented:
Hey @MillhioreBT, did you had time to test it?
